packages/platform/src/editor/Editor.tsx

```diff
@@ -80,24 +88,46 @@ function Placeholder(): JSX.Element {
  * @param props.logger - Logger instance.
  * @returns the editor element.
  */
-export default function Editor<TLogger extends LoggerBasic>({
-  usj,
-  viewOptions,
-  scrRef,
-  setScrRef,
-  nodeOptions,
-  isReadonly,
-  onChange,
-  logger,
-}: EditorProps<TLogger>): JSX.Element {
+const Editor = forwardRef(function Editor<TLogger extends LoggerBasic>(
+  {
+    defaultUsj,
+    viewOptions,
+    scrRef,
+    setScrRef,
+    nodeOptions,
+    isReadonly,
+    onChange,
+    logger,
+  }: EditorProps<TLogger>,
+  ref: React.ForwardedRef<EditorRef>,
+): JSX.Element {
+  const editorRef = useRef<LexicalEditor>(null);
+  const [usj, setUsj] = useState(defaultUsj);
+  const [loadedUsj, , setEditedUsj] = useDeferredState(usj);
```

Comment:
Do we still need this if we stop setting the USJ up above in the `onChange`? I appreciate that deep equaling the state is nice to avoid having to rerender the editor unnecessarily. However, this pattern seems to deviate somewhat significantly from normal React patterns and thereby increase mental complexity of this component. It also seems to add a render and a duplicate of this state (probably neither of which are huge deals; just considering the consequences). If it is still very important to do a deep equals check even after making the component uncontrolled (meaning we can likely assume people won't be setting this state as a result of the `onChange` running unless they're doing it wrong), could we do that somewhere in a `memo` function? Maybe in `UpdateStatePlugin`? Maybe there are too many other props in there; dunno. Just a thought.

Reply:
No, we don't need it if we use the uncontrolled component properly. This still needs some work however when you update USX in P.B it currently spits out a new USX value so we need this logic either in the editor component or in every extension that uses it (or in the USX PDP). I elected to put it in the editor for now but expect to remove it later.